### ISSUE TITLE
Integrate the new scons-based support for Spartan

### ISSUE THREAD

**Issue by msprotz** [project-everest/everest-ci#21]
Several sub-tasks:
- [x] add Python2.7 in the Windows build environment _and_ in the Linux one. Note: we most likely want the native Windows Python, not the Cygwin one.
- [x] adapt the everest script to check for Python (non-Windows), and check for Python
- [ ] _and_ offer to install it (Windows)
- [x] adapt the everest script to change the CMD invocation for Spartan at https://github.com/project-everest/everest/blob/master/everest#L522 with the one that will trigger Scons from a Native x64 VS Tools Prompt (the one that the Spartan folks are using) possibly skipping scons.bat since it's a one-liner after all
- [x] test the whole setup using the ubuntu-from-scratch scenario (depends on Spartan moving to GitHub first)
- [ ] if Spartan drops Scons from their source tree, need to install scons on the windows machine too -- need to make the script download & launch the scons installer in quiet mode, or something similar (see http://scons.org/pages/download.html)

CC @barrybo

**Comment by msprotz** [comment]
- My commit https://github.com/project-everest/everest/commit/8cd00479e82bcff5f6db34d9cea25072f9ba47f0 fixes "adapt the everest script to check" and "change the CMD invocation"
- My commit https://github.com/project-everest/everest-ci/commit/d0ff7e6b9c54713ed7a74b8d8fc1e758ae0ea021 adds scons into the Linux build environment

**Comment by vkanne-msft** [comment]
Waiting on VALE move to github

**Comment by vkanne-msft** [comment]
Vale is now on GitHub. Can we update our scripts to use the GitHub repo?

**Comment by darrenge** [comment]
I would think so ... first thing is to get "green" in all our existing tests.

**Comment by BarryBo** [comment]
I successfully cloned and built using the instructions on the project homepage (set up fslexyacc/python27/scons and used my VS2015 tools).  I built Windows 32-bit x86 only.

**Comment by BarryBo** [comment]
... and scons... the gzip tarball works on Windows.  There is no need to use the Windows Installer .exe to install it.

**Comment by msprotz** [comment]
https://github.com/project-everest/everest/tree/vale is a first attempt at this -- happy for anyone to pick it up and/or merge into master

as far as I know the everest build & test has been green for the past few builds

**Comment by sishtiaq** [comment]
@darrenge Done? Can we close this?

**Comment by darrenge** [comment]
I am not sure where we are at with this one.

**Comment by msprotz** [comment]
I checked a couple checkboxes -- ideally it'd be nice if the Everest script offered to install scons on Windows via the msi

**Comment by sishtiaq** [comment]
Ah, I see. I didn't see that unticked box.

**Comment by darrenge** [comment]
Looking into getting Python installed and scons.bat installed automatically.

**Comment by darrenge** [comment]
Put in a pull request to install Python 2.7 and Scons on the Windows side of things if they are not already installed. Once that merges, then this issue can be closed.

**Comment by msprotz** [comment]
Closed via https://github.com/project-everest/everest/pull/30 -- thanks much!
